### ISSUE TITLE
Add NN_baseline (simple CNN, U-Net)

### ISSUE THREAD

**Issue by xp1632** [xp1632/Aalen_working_log#69]
[Link](https://claude.ai/chat/2aa9173d-8f2f-421d-91e5-88e8c7f46204 )
- [Dataset generator] for generating [n] pairs of images with the same OP setting for training NN

<img width="986" height="791" alt="Image" src="https://github.com/user-attachments/assets/c0e48032-5ee5-4522-b528-f245be84e8ef" />

```
# Example 1: Tier 1 - simple point-wise
python generate_nn_dataset.py \
    --name tier1_point \
    --ops "DiffBrightness:0.8,DiffGamma:1.2" \
    --num_pairs 10

# Example 2: Tier 2 - local operations (future)
python generate_nn_dataset.py \
    --name tier2_local \
    --ops "GaussianBlur:1.5,Sharpen:0.3" \
    --num_pairs 15

# Example 3: Tier 3 - combined (future)
python generate_nn_dataset.py \
    --name tier3_combined \
    --ops "DiffBrightness:0.9,GaussianBlur:1.0,DiffGamma:1.1" \
    --num_pairs 20
```

**Comment by xp1632** [comment]
Plan:

<img width="771" height="771" alt="Image" src="https://github.com/user-attachments/assets/d4d2c82f-3e0c-4c83-82ca-54c7a98f4b7c" />

**Comment by xp1632** [comment]
- Simple 3-layer CNN


  SimpleCNN                                                  
  ├── Conv2d(3→32) → ReLU                                    
 ├── Conv2d(32→32) → ReLU                                   
  └── Conv2d(32→3) → Sigmoid                                 
 Total: ~11,000 parameters                                  

---

  - Workflow                                                   
                                                             
  Dataset (10 pairs)                                         
       │                                                     
       ├── Train (N pairs) ──► Train CNN ──► Save model      
       │                                                     
       └── Test (10-N pairs) ──► Evaluate ──► PSNR results   
---

- Result
```
NN Baseline Experiment
============================================================

Dataset: C:\Users\06610\Workspace\my-chatbox-extension\my_chatbox_extension\eval\nn_datasets\tier1_test
Total pairs: 10
Train pairs: 1 (['NN01'])
Test pairs:  9 (['NN02', 'NN03', 'NN04', 'NN05', 'NN06', 'NN07', 'NN08', 'NN09', 'NN10'])

Training for 1000 epochs...
  Epoch  100/1000: loss = 0.000453
  Epoch  200/1000: loss = 0.000320
  Epoch  300/1000: loss = 0.000207
  Epoch  400/1000: loss = 0.000157
  Epoch  500/1000: loss = 0.000120
  Epoch  600/1000: loss = 0.000106
  Epoch  700/1000: loss = 0.000088
  Epoch  800/1000: loss = 0.000082
  Epoch  900/1000: loss = 0.000069
  Epoch 1000/1000: loss = 0.000063

Evaluating on test set...

============================================================
RESULTS
============================================================
Training pairs: 1
Test pairs:     9
Final loss:     0.000063
Train PSNR:     41.98 dB
Test PSNR:      30.12 ± 3.12 dB
------------------------------------------------------------
Per-image results:
  NN02: 37.39 dB
  NN03: 29.72 dB
  NN04: 26.20 dB
  NN05: 32.97 dB
  NN06: 28.68 dB
  NN07: 27.74 dB
  NN08: 28.23 dB
  NN09: 30.15 dB
  NN10: 30.03 dB
============================================================

Results saved to C:\Users\06610\Workspace\my-chatbox-extension\my_chatbox_extension\eval\nn_datasets\tier1_test\nn_results_n1.json
============================================================
NN Baseline Experiment
============================================================

Dataset: C:\Users\06610\Workspace\my-chatbox-extension\my_chatbox_extension\eval\nn_datasets\tier1_test
Total pairs: 10
Train pairs: 5 (['NN01', 'NN02', 'NN03', 'NN04', 'NN05'])
Test pairs:  5 (['NN06', 'NN07', 'NN08', 'NN09', 'NN10'])

Training for 1000 epochs...
  Epoch  100/1000: loss = 0.000595
  Epoch  200/1000: loss = 0.000325
  Epoch  300/1000: loss = 0.000263
  Epoch  400/1000: loss = 0.000237
  Epoch  500/1000: loss = 0.000170
  Epoch  600/1000: loss = 0.000135
  Epoch  700/1000: loss = 0.000107
  Epoch  800/1000: loss = 0.000130
  Epoch  900/1000: loss = 0.000221
  Epoch 1000/1000: loss = 0.000074

Evaluating on test set...

============================================================
RESULTS
============================================================
Training pairs: 5
Test pairs:     5
Final loss:     0.000074
Train PSNR:     41.89 dB
Test PSNR:      37.14 ± 2.53 dB
------------------------------------------------------------
Per-image results:
  NN06: 36.87 dB
  NN07: 34.79 dB
  NN08: 34.07 dB
  NN09: 40.61 dB
  NN10: 39.35 dB
============================================================
```

**Comment by xp1632** [comment]
Add NN comparsion_Evaluation v3
https://claude.ai/chat/2aa9173d-8f2f-421d-91e5-88e8c7f46204

- Added SSIM and better worklfow
- model.py contains simpleCNN and Unet

and in Teams repo:
https://github.com/mlcvAalen/Team_meeting/issues/35

<img width="1090" height="7100" alt="Image" src="https://github.com/user-attachments/assets/a7499f77-87bd-4f37-b904-0b191ab83850" />

**Comment by xp1632** [comment]
### Add Norm Layer to original Unet due to turbulence performance:

We use U-Net architecture in  (Ronneberger et al., 2015) , but noticed performance 

with batch normalization (Ioffe & Szegedy, 2015 )for stable training.

-----
And it becomes even worse, so we use 3 seed runs to smooth:
